autoload/selector.vim

```diff
@@ -201,30 +238,102 @@ function! selector#OpenWindow(infolist, ...) abort
 
   " Open the window in the specified window position.  Typically, this opens up
   " a flat window on the bottom (as with split).
-  execute l:window_position . ' ' . l:win_size . 'new'
-  call s:SetWindowOptions()
-  silent execute 'file ' . l:window_name
-  let [l:lines, l:data] = s:SplitLinesAndData(a:infolist)
+  execute l:position l:win_size 'new'
+  let s:selectors_by_bufnum[bufnr('%')] = self
+  call s:SetWindowOptions(self)
+  silent execute 'file' self._name
+  let [l:lines, l:data] = s:SplitLinesAndData(self._infolist)
   let b:selector_lines_data = l:data
-  call s:InstantiateKeyMaps()
+  call s:InstantiateKeyMaps(self._mappings)
   setlocal noreadonly
   setlocal modifiable
-  let s:verbose_help = 0
   call maktaba#buffer#Overwrite(1, line('$'), l:lines)
   " Add the help comments at the top (do this last so cursor stays below it).
-  call append(0, s:GetHelpLines(0))
+  call append(0, self._GetHelpLines())
   setlocal readonly
   setlocal nomodifiable
 
   " Restore the previous windows view
-  let buffer_window = winnr()
+  let l:buffer_window = winnr()
   call selector#ReturnToWindow()
   call winrestview(s:current_savedview)
-  execute buffer_window  . 'wincmd w'
+  execute l:buffer_window  'wincmd w'
+
+  return self
+endfunction
+
+
+""
+" @private
+" Gets data associated with {lineno}, as passed in 2-item form of infolist when
+" creating a selector with @function(#Create).
+" @throws NotFound if no data was configured for requested line.
+function! selector#DoGetLineData(lineno) dict abort
+  let l:lineno = a:lineno - len(self._GetHelpLines())
+  if has_key(b:selector_lines_data, l:lineno)
+    return b:selector_lines_data[l:lineno]
+  endif
+  throw maktaba#error#NotFound('Associated data for selector line %d', l:lineno)
+endfunction
+
+
+""
+" @public
+" @usage {infolist} [ResetMapper] [window_name] [window_position]
+" WARNING: This is a legacy function and will soon be deprecated and removed.
+" Open a selector window named [window_name] based on {infolist}, a list of
```

Comment:
That's just vim help syntax (`:h help-writing`). {} is for the actual function arguments. \`\` is for other commands or syntax-y things.